src/polyfill.js

```diff
@@ -0,0 +1,33 @@
+const React = require('react');
+const reactHooks = require('./withHooks');
+
+const useNative = !!React.useState;
+
+const { default: withHooks, ...hooks } = reactHooks;
+
+const createElementWithHooks = (() => {
+  const componentMap = new Map();
+  return (el, props, ...children) => {
+    if (typeof el === 'function' && el.prototype && !el.prototype.isReactComponent) {
```

Comment:
Just realised that `typeof (() => {}) === 'function'` but `(() => {}).prototype === undefined`, so...

```suggestion
    if (typeof el === 'function' && !el.prototype || (el.prototype && !el.prototype.isReactComponent)) {
```

Reply:
I'm using [`react-is`](https://www.npmjs.com/package/react-is) to detect element types and porting the official tests of hooks to this repo.